internal/service/dataexchange/event_action_test.go

```diff
@@ -0,0 +1,671 @@
+			{
+				Config: testAccEventActionConfig_encryption(bucketName, dataSetId, *identity.Account),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckEventActionExists(ctx, resourceName, &dataexchange.GetEventActionOutput{}),
+					resource.TestCheckResourceAttr(resourceName, "action_export_revision_to_s3.s3_encryption_type", "AES256"),
+					acctest.MatchResourceAttrRegionalARN(ctx, resourceName, names.AttrARN, "dataexchange", regexache.MustCompile(`event-actions/.+`)),
+				),
+			},
+		},
+	})
+}
+
+func TestAccDataExchangeEventAction_kmsKeyEncryption(t *testing.T) {
+	ctx := acctest.Context(t)
+	resourceName := "aws_dataexchange_event_action.test"
+	bucketName := strconv.Itoa(int(time.Now().UnixNano()))
+	if _, okAcc := os.LookupEnv("TF_ACC"); !okAcc {
+		t.Skipf("TF_ACC must be set")
+	}
+
+	dataSetId, err := helperAccEventActionGetReceivedDataSet(ctx, awstypes.AssetTypeS3Snapshot)
+
+	cfg, err := config.LoadDefaultConfig(ctx)
+	if err != nil {
+		t.Error(err)
+	}
+	stsConn := sts.NewFromConfig(cfg)
+	identity, err := stsConn.GetCallerIdentity(ctx, &sts.GetCallerIdentityInput{})
+	if err != nil {
+		t.Error(err)
+	}
+
+	if dataSetId == "" {
+		return
+	}
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { acctest.PreCheck(ctx, t); acctest.PreCheckPartitionHasService(t, names.DataExchangeEndpointID) },
+		ErrorCheck:               acctest.ErrorCheck(t, names.DataExchangeServiceID),
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories,
+		CheckDestroy:             testAccCheckEventActionDestroy(ctx),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccEventActionConfig_kmsKeyEncryption(bucketName, dataSetId, *identity.Account),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckEventActionExists(ctx, resourceName, &dataexchange.GetEventActionOutput{}),
+					resource.TestCheckResourceAttr(resourceName, "action_export_revision_to_s3.s3_encryption_type", "aws:kms"),
+					acctest.MatchResourceAttrRegionalARN(ctx, resourceName, names.AttrARN, "dataexchange", regexache.MustCompile(`event-actions/.+`)),
+				),
+			},
+		},
+	})
+}
+
+func testAccEventActionPreCheck(ctx context.Context, t *testing.T) {
+	conn := acctest.Provider.Meta().(*conns.AWSClient).DataExchangeClient(ctx)
+
+	input := &dataexchange.ListEventActionsInput{}
+
+	_, err := conn.ListEventActions(ctx, input)
+
+	if acctest.PreCheckSkipError(err) {
+		t.Skipf("skipping acceptance testing: %s", err)
+	}
+	if err != nil {
+		t.Fatalf("unexpected PreCheck error: %s", err)
+	}
+}
+
+func testAccCheckEventActionDestroy(ctx context.Context) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		conn := acctest.Provider.Meta().(*conns.AWSClient).DataExchangeClient(ctx)
+
+		for _, rs := range s.RootModule().Resources {
+			if rs.Type != "aws_dataexchange_event_action" {
+				continue
+			}
+
+			_, err := conn.GetEventAction(ctx, &dataexchange.GetEventActionInput{
+				EventActionId: aws.String(rs.Primary.ID),
+			})
```

Comment:
The finder function `findEventActionByID` can be used here in place of the call directly to the Get API. The same is true of the check for existence below.